k2/csrc/ragged_ops.cu

```diff
@@ -1144,6 +1145,27 @@ Array1<int32_t> GetTransposeReordering(Ragged<int32_t> &src, int32_t num_cols) {
 
   K2_CHECK_EQ(device_type, kCuda);
 
+#if 1
+  const int32_t *values_data = src.values.Data();
+  int32_t num_buckets = num_cols;
+  int32_t num_elements = src.values.Dim();
+
+  Array1<int32_t> ans = Range(context, num_elements, 0);
+  auto lambda = [values_data, num_elements] __device__(uint32_t i) -> uint32_t {
+    return i < num_elements ? values_data[i] : 0;
```

Comment:
@csukuangfj just a question: why can `i` equal to `num_elements` here? cudpp will pass `i` greater than `num_elements`?

Reply:
Yes, you can print what `i` looks like and you will get lots of very large numbers, which result in segfault if you remove the condition statement.

Reply:
OK, thanks, I guess it may pass some thread Ids to this lambda function instead of element index.